### PR TITLE
TF-733 [Web] Can change the font/color/strike text to the body in composer

### PR DESCRIPTION
**Pull request by dab246**
### Result Demo
https://user-images.githubusercontent.com/80730648/180176528-3f587595-0a2a-4b2a-9932-cce7f6c0d6c1.mov

#### Supported FontFamily

https://user-images.githubusercontent.com/80730648/180199405-b7cee7ab-fd68-4082-b343-8878872608b5.mov

### CODE REVIEW

File: lib/features/composer/presentation/model/font_name_type.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+enum FontNameType {
+  sansSerif,
+  times,
+  courier;
```
Comment (hoangdat) [inline lib/features/composer/presentation/model/font_name_type.dart:5]: three fonts family are supported by the library? only three? if we want to support more, how to add?

File: lib/features/composer/presentation/controller/rich_text_web_controller.dart [hunk 2]
```diff
@@ -86,4 +158,12 @@ class RichTextWebController extends GetxController {
     log('RichTextWebController::refactorContentHasInlineImage(): listInlineAttachment: $listInlineAttachment');
     return Tuple2(newContent, listInlineAttachment);
   }
+
+  void appNewFontStyle(FontNameType? newFont) {
```
Comment (hoangdat) [inline lib/features/composer/presentation/controller/rich_text_web_controller.dart:162]: apply*, right?

File: lib/features/composer/presentation/controller/base_rich_text_controller.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        context,
+        currentColor,
+        title: AppLocalizations.of(context).chooseAColor,
+        textActionSetColor: AppLocalizations.of(context).setColor,
+        textActionResetDefault: AppLocalizations.of(context).resetToDefault,
+        textActionCancel: AppLocalizations.of(context).cancel,
+        cancelActionCallback: () => popBack(),
+        resetToDefaultActionCallback: () {
+          onResetToDefault?.call();
+          popBack();
```
Comment (hoangdat) [inline lib/features/composer/presentation/controller/base_rich_text_controller.dart:28]: Base on the demo, it seems not succeeded

Comment (dab246) [reply lib/features/composer/presentation/controller/base_rich_text_controller.dart:28]: Please check your eyes again. Let's watch it one more time

File: lib/features/composer/presentation/model/font_name_type.dart (hunk 1 above)
Comment (dab246) [reply lib/features/composer/presentation/model/font_name_type.dart:5]: - Simply add `fontFamily` that web supported. If you want to add some other font you have to include a link in the `head` tag of the html for it. Ex:
```html
<head>
<link href='https://fonts.googleapis.com/css?family=Sofia' rel='stylesheet'>
<style>
body {
     font-family: 'Sofia';font-size: 22px;
}
</style>
</head>
```
You can read more here [https://www.w3schools.com/cssref/css_websafe_fonts.asp](https://www.w3schools.com/cssref/css_websafe_fonts.asp)

File: lib/features/composer/presentation/controller/rich_text_web_controller.dart (hunk 2 above)
Comment (dab246) [reply lib/features/composer/presentation/controller/rich_text_web_controller.dart:162]: ok

**Review by hoangdat — APPROVED** [review]

**Pull request merged**